### PR TITLE
Gatkcnvcaller

### PR DESCRIPTION
**Pull request by ramprasadn**
## PR checklist

This PR adds GATK's germline CNV calling workflow to the pipeline, and includes relevant changes in the documentation.  

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/raredisease/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, also make a PR on the nf-core/raredisease _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Ensure the test suite passes (`nextflow run . -profile test_one_sample,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: conf/modules/prepare_references.config [hunk 1]
```diff
@@ -117,4 +117,15 @@ process {
...
         ]
     }
+
+    withName: '.*PREPARE_REFERENCES:GATK_PREPROCESS_WGS' {
+        ext.args =  { "--padding 0  --interval-merging-rule OVERLAPPING_ONLY --exclude-intervals ${params.mito_name}" }
+        ext.when = { params.analysis_type.equals("wgs") && !params.readcount_intervals }
+    }
+
+    withName: '.*PREPARE_REFERENCES:GATK_PREPROCESS_WES' {
+        ext.args   = { "--bin-length 0  --interval-merging-rule OVERLAPPING_ONLY --exclude-intervals ${params.mito_name}" }
```
Comment (jemten) [inline conf/modules/prepare_references.config:127]:
```suggestion
        ext.args   = { "--bin-length 0 --interval-merging-rule OVERLAPPING_ONLY --exclude-intervals ${params.mito_name}" }
```

File: conf/modules/prepare_references.config [hunk 2]
```diff
@@ -117,4 +117,15 @@ process {
             enabled: false
         ]
     }
+
+    withName: '.*PREPARE_REFERENCES:GATK_PREPROCESS_WGS' {
+        ext.args =  { "--padding 0  --interval-merging-rule OVERLAPPING_ONLY --exclude-intervals ${params.mito_name}" }
```
Comment (jemten) [inline conf/modules/prepare_references.config:122]:
```suggestion
        ext.args =  { "--padding 0 --interval-merging-rule OVERLAPPING_ONLY --exclude-intervals ${params.mito_name}" }
```

File: docs/output.md [hunk 3]
```diff
@@ -10,6 +10,8 @@ The directories listed below will be created in the results directory after the
 
 The pipeline is built using [Nextflow](https://www.nextflow.io/) and processes data using the following steps:
 
+#### GATK GermlineCNVCaller (CNV calling)
+
```
Comment (jemten) [inline docs/output.md:14]:
```suggestion
```

File: subworkflows/local/variant_calling/call_sv_germlinecnvcaller.nf [hunk 4]
```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env nextflow
```
Comment (jemten) [inline subworkflows/local/variant_calling/call_sv_germlinecnvcaller.nf:1]: Is it the new standard to add a shebang for nextflow subworkflows? We should add it to the subworkflows as well then (another PR).

**Review by jemten — APPROVED** [review]
Very nice and well written!

File: subworkflows/local/variant_calling/call_sv_germlinecnvcaller.nf [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+    main:
+        ch_versions = Channel.empty()
+
+        input = ch_bam_bai.combine( ch_readcount_intervals.collect{ it[1] } )
+
+        GATK4_COLLECTREADCOUNTS ( input, ch_fasta, ch_fai, ch_genome_dictionary )
+
+        GATK4_COLLECTREADCOUNTS.out.tsv
+                .groupTuple()
```
[the remark this answers is not in the document]
Comment (ramprasadn) [reply subworkflows/local/variant_calling/call_sv_germlinecnvcaller.nf]: Its actually not necessary here.. I have removed it now.

File: subworkflows/local/variant_calling/call_sv_germlinecnvcaller.nf (hunk 4 above)
Comment (ramprasadn) [reply subworkflows/local/variant_calling/call_sv_germlinecnvcaller.nf]: Nope.. removed it.

**Pull request merged**